plugins/common/compiler/api.ts

```diff
@@ -0,0 +1,16 @@
+import { CompilationFileSources, CompilationResult } from './type';
+
+export interface CompilerApi {
+  events: {
+    compilationFinished: (
+      fileName: string,
+      source: CompilationFileSources,
+      languageVersion: string,
+      data: CompilationResult
+    ) => void
+  }
+  methods: {
+    getCompilationResult(): CompilationResult
```

Comment:
I'd be perhaps be more flexible with the `CompilationResult` type or at least renaming it EVMCompilerAPI.
We are just about to releasing a pluginZokrates compiler and it's not EVM based.
Also basically, for evm compiler input and output should comply with https://solidity.readthedocs.io/en/v0.5.11/using-the-compiler.html#compiler-input-and-output-json-description

Reply:
So you want "compiler" to go beyond EVM based output ? The current types follow the documentation from solidity.

Reply:
renaming compilerAPI => EVMCompilerAPI is fine